mlflow/xgboost/__init__.py

```diff
@@ -361,8 +362,31 @@ def predict(
         Returns:
             Model predictions.
         """
+        import xgboost as xgb
+
         predict_fn = _wrapped_xgboost_model_predict_fn(self.xgb_model)
-        return predict_fn(dataframe, **(params or {}))
+        params = params or {}
+        # filter is applied inside predict_fn wrapper for xgb.Booster
+        if not isinstance(self.xgb_model, xgb.Booster):
+            params = _filter_allowed_kwargs(predict_fn, params)
+        return predict_fn(dataframe, **params)
+
+
+def _filter_allowed_kwargs(predict_fn, kwargs):
+    filtered_kwargs = {}
+    allowed_params = inspect.signature(predict_fn).parameters
```

Comment:
Dump question (sorry if it's already answered in discussion, I was a bit lost during stand-up) - why do we want to allow signature to contain unsupported param for the model? Doesn't it make signature useless?

Reply:
I think that's because it's challenging to infer supported parameters? Imagine you have a model whose `predict` looks like this:

```python
def predict(self, *args, **kwargs):
    ...
```

Reply:
But the issue is the signature contains extra params, not the case where we fail to infer some?

Reply:
Also we use inspect.signature in the logic... if it is reliable, why not doing it for inferring param signature?

Reply:
Personally I don't think we should do this, but this does cause feature store's regression, which is a bug they should fix. But considering this is a regression, if we can detect invalid params then we can avoid some user errors. 

The case when predict function include args or kwargs, this will cause error by excluding all of them, but for xgboost models seems they don't do this, they don't accept kwargs, they use named parameters inside predict functions.

Reply:
Ahhh got it, thanks for the clarification. Can we note those decision reasons as comments, so that (1) we can confidently remove this once they fix the bug (2) we can avoid accidentally updating the logic without knowing the FS dependency?

Reply:
Added comment!